src/matchbox/server/api.py

```diff
@@ -76,18 +92,49 @@ async def clear_backend():
 
 
 @app.get("/sources")
-async def list_sources():
-    raise HTTPException(status_code=501, detail="Not implemented")
+async def list_sources(
+    backend: Annotated[MatchboxDBAdapter, Depends(get_backend)],
+) -> Sources:
+    datasets = backend.datasets.list()
+    result = []
+    for dataset in datasets:
+        result.append(
+            SourceItem(
+                table=dataset.table,
+                id=dataset.id,
+                schema=dataset.schema,
+                resolution=hexlify(dataset.resolution).decode("ascii"),
+            )
+        )
+    return Sources(sources=result)
 
 
 @app.get("/sources/{hash}")
-async def get_source(hash: str):
-    raise HTTPException(status_code=501, detail="Not implemented")
-
-
-@app.post("/sources/{hash}")
-async def add_source(hash: str):
-    raise HTTPException(status_code=501, detail="Not implemented")
+async def get_source(
+    hash: str, backend: Annotated[MatchboxDBAdapter, Depends(get_backend)]
+) -> dict[str, SourceItem] | str:
+    datasets = backend.datasets.list()
+    for dataset in datasets:
+        resolution = hexlify(dataset.resolution).decode("ascii")
+        if resolution == hash:
+            result_obj = SourceItem(
+                table=dataset.table,
+                id=dataset.id,
+                schema=dataset.schema,
+                resolution=resolution,
+            )
+            return {"source": result_obj}
+    return "Source not found"
```

Comment:
if the source is not found we need to return a 404. And it is found, let's just return the object, no need to wrap it in a dict.